src/index.ts

```diff
@@ -1,923 +1,1064 @@
+
+    // Envoy - New Hosted Agent Sign-up Sequence
+    this.ws_server.register("holo/agent/signup", async ([hha_hash, agent_id]) => {
+      log.normal("Received sign-up request from Agent (%s) for HHA ID: %s", agent_id, hha_hash)
+
+      const failure_response = (new HoloError("Failed to create a new hosted agent")).toJSON();
+      let resp;
+
+      log.info("Retrieve the holo-hosting-app cell id using the Installed App Id: '%s'", HHA_INSTALLED_APP_ID);
+      const appInfo = await this.callConductor("internal", { installed_app_id: HHA_INSTALLED_APP_ID });
+
+      if (!appInfo) {
+        log.error("Failed during HHA AppInfo lookup: %s", appInfo);
+        return failure_response;
+      }
+
+      const hha_cell_id = appInfo.cell_data[0][0];
+      const buffer_host_agent_hha_id = hha_cell_id[1];
+
+      log.info("Look-up Hosted App's HHA record using ID '%s'", hha_hash);
+      resp = await this.callConductor("internal", {
+        "cell_id": hha_cell_id,
+        "zome_name": "hha",
+        "fn_name": "get_happ",
+        "payload": hha_hash,
+        "cap": null,
+        "provenance": buffer_host_agent_hha_id,
+      });
+
+      if (!resp) {
+        log.error("Failed during App Details lookup in HHA: %s", resp);
+        return failure_response;
+      }
+      const app = resp;
+      const installed_app_id = `${hha_hash}:${agent_id}`;
+
+      log.silly("HHA bundle: %s", app);
+      // Example response
+      // {
+      // 		happ_id: HeaderHash,
+      // 		happ_bundle: {
+      //			hosted_url: String,
+      //			happ_alias: String,
+      //			ui_path: String,
+      //			name: String,
+      //			dnas: [{
+      //				hash: String, // hash of the dna, not a stored dht address
+      //				path: String,
+      //				nick: Option<String>, << make this required in hha!
+      //			}],
+      //		},
+      // 		provider_pubkey: AgentPubKey,
+      // }
+
+      log.info("Found %s DNA(s) for the app bundle with HHA ID: %s", app.happ_bundle.dnas.length, hha_hash);
+
+      const buffer_agent_id = Codec.AgentId.decodeToHoloHash(agent_id);
+      log.info("Encoded Agent ID (%s) into buffer form: %s", agent_id, buffer_agent_id);
+
+      let failed = false;
+      try {
+        let adminResponse;
+        // - Install App - This admin function creates cells for each dna with associated nick, under the hood.
+        try {
+          log.info("Installing App with HHA ID (%s) as Installed App ID (%s) ", hha_hash, installed_app_id);
+
+          let dnas;
+          if (this.opts.hosted_app!.dnas && this.opts.mode === Envoy.DEVELOP_MODE) {
+            dnas = this.opts.hosted_app.dnas;
+          }
+
+          adminResponse = await this.callConductor("master", 'installApp', {
+            installed_app_id,
+            agent_key: buffer_agent_id,
+            dnas: dnas || app.happ_bundle.dnas.map(dna => {
+              let nick;
+              if (!dna.nick) {
+                const dnaFileName = dna.path.split("/");
```

Comment:
This is where we need to download the DNA from the URL and not depend on a path.

Reply:
Agreed and already done.